libazureinit/src/provision/ssh.rs

```diff
@@ -37,17 +39,80 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
+pub(crate) fn get_authorized_keys_path_from_sshd() -> io::Result<Option<PathBuf>>
+{
+    let sshd_output = Command::new("sshd").arg("-G").output()?;
+
+    if !sshd_output.status.success() {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!("sshd -G failed with status: {}", sshd_output.status),
+        ));
+    }
+
+    let stdout = sshd_output.stdout;
+    let sshd_output = String::from_utf8_lossy(&stdout);
+    for line in sshd_output.lines() {
+        if line.starts_with("authorizedkeysfile") {
+            let keypath: Vec<&str> = line.split_whitespace().collect();
+            if keypath.len() > 1 {
+                return Ok(Some(PathBuf::from(keypath[1])));
```

Comment:
Please add a tracing call to output the path that we use

Reply:
Done